gems/aws-sdk-s3/lib/aws-sdk-s3/plugins/s3_signer.rb

```diff
@@ -199,15 +199,23 @@ def build_v4_signer(options = {})
           end
 
           def new_hostname(context, region)
-            bucket = context.params[:bucket]
-            if region == 'us-east-1'
-              "#{bucket}.s3.amazonaws.com"
```

Comment:
I think this qualifies as "old code". I've removed it because we already force regionalization in EndpointProvider here: https://github.com/aws/aws-sdk-ruby/blob/master/gems/aws-partitions/lib/aws-partitions/endpoint_provider.rb#L88